nextflow_schema.json

```
@@ -303,6 +303,22 @@
                     "fa_icon": "fas fa-water",
                     "hidden": false,
                     "default": 25
+                },
+                "juicer_jvm_params": {
+                    "type": "string",
+                    "description": "JVM heap parameters for juicer",
+                    "default": "-Xms512m -Xmx4096m",
+                    "fa_icon": "far fa-clock",
+                    "hidden": false,
+                    "help_text": "For 64bit system, you may want to set the Xmx > 4g, eg. '-Xms1024m -Xmx32g'"
+                },
+                "juicer_tools_jar": {
+                    "type": "string",
+                    "description": "The juicer_tools path",
+                    "default": null,
```

If the default is `null` then we shouldn't include it in the schema.

Hmm, was this automatically put in by `nf-core schema build`? If so then that's a bug and we need to fix that.

---

If changing the config as suggested above:

```suggestion
                    "default": "https://github.com/aidenlab/Juicebox/releases/download/v2.13.06/juicer_tools_2.13.06.jar",
```

---

No. I manually edit it. I found that the nf-core tools will always remove the last newline. And later I give up to edit it by tools. 
I will test the download on fly to make sure it will work properly. Thank you for clarify it again.

---

Ah ok - and the newline is annoying because it breaks the linting or something? I think I might have hit this myself before in the past. We should be able to solve this.